editor/src/messages/layout/layout_message_handler.rs

```diff
@@ -208,55 +239,57 @@ impl<F: Fn(&MessageDiscriminant) -> Vec<KeysGroup>> MessageHandler<LayoutMessage
 }
 
 impl LayoutMessageHandler {
+	/// Diff the update and send to the frontend where necessary
+	fn send_layout(&mut self, layout_target: LayoutTarget, new_layout: Layout, responses: &mut VecDeque<Message>, action_input_mapping: &impl Fn(&MessageDiscriminant) -> Vec<KeysGroup>) {
+		// We don't diff the menu bar layout yet.
+		if layout_target == LayoutTarget::MenuBar {
```

Comment:
I would not check layout target but rather by layout type. `MenuLayout` vs `WidgetLayout`

```rust
pub enum Layout {
	WidgetLayout(WidgetLayout),
	MenuLayout(MenuLayout),
}
```